lib/main.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2012-2016 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+var _ = require('lodash');
+var bedrock = require('bedrock');
+var config = bedrock.config;
+var brPermission = require('bedrock-permission');
+var brPassport = require('bedrock-passport');
+
+var ensureAuthenticated = brPassport.ensureAuthenticated;
+
+require('bedrock-express');
+
+bedrock.events.on('bedrock-express.configure.routes', function(app) {
+
+  // Permission routes -----------------------------------------------------
+
+  /*
+    Retrieves all permissions from the config held permissions.
+   */
+  app.get('/permissions', ensureAuthenticated, function(req, res, next) {
+    var permissions = config.permission.permissions;
+    res.status(200).json(permissions);
+  });
+
+  // Role routes ------------------------------------------------------------
+  
+  /*
+    Gets all existing role
+  */
+  app.get('/roles', ensureAuthenticated, function(req, res, next) {
+    brPermission.getRoles(req.user.identity, function(err, result) {
+      if (err) {
+        return next(err);
+      }
+      var roles = _.map(result, function(result) {
+        return result.role;
+      });
+      res.status(200).json(roles);
+    });
+  });
+
+  /**
+    Inserts a newly created role to the database
+    
+    @param role role that is to be added to store.
+
+    @return role that is stored.
+  */
+  app.post('/roles', ensureAuthenticated, function(req, res, next) {
+    var role = req.body.role;
+    brPermission.addRole(req.user.identity, role, function(err, result) {
+      if(err || result === undefined) {
+        return next(err);
+      } else {
+        res.status(201).json(result.role);
+      }
+    });
+  });
+
+  /*
+    Edits a role that is already in the database
+
+    @param role new role that is to replace the previous role with the same ID.
+
+    @return resulting updated role that is stored.
+  */
+  app.put('/roles/:id',ensureAuthenticated, function(req, res, next) {
+    var role = req.body.role;
+    brPermission.updateRole(req.user.identity, role, function(err, result) {
+      if(err || result === undefined) {
+        return next(err);
+      } else {
+        res.status(202).json(result);
```

Comment:
This should probably just be `res.json(result)` and use default 200.  202 is more for delayed background processing of a request but in this case it is processed immediately.